neqo-transport/src/connection/tests/handshake.rs

```diff
@@ -1194,3 +1195,63 @@ fn emit_authentication_needed_once() {
     _ = client.process(server2.as_dgram_ref(), now());
     assert_eq!(0, authentication_needed_count(&mut client));
 }
+
+#[test]
+fn client_initial_retransmits_identical() {
+    let mut now = now();
+    let mut client = default_client();
+
+    // Force the client to retransmit its Initial packet a number of times and make sure the
+    // retranmissions are identical to the original. Also, verify the PTO durations.
+    for i in 1..=5 {
+        let ci = client.process(None, now).dgram().unwrap();
+        assert_eq!(ci.len(), client.plpmtu());
+        assert_eq!(
+            client.stats().frame_tx,
+            FrameStats {
+                crypto: i,
+                all: i,
+                ..Default::default()
+            }
+        );
+        let pto = client.process(None, now).callback();
+        assert_eq!(pto, DEFAULT_RTT * 3 * (1 << (i - 1)));
+        now += pto;
+    }
+}
+
+#[test]
+fn server_initial_retransmits_identical() {
+    let mut now = now();
+    let mut client = default_client();
+    let mut ci = client.process(None, now).dgram();
+
+    // Force the server to retransmit its Initial packet a number of times and make sure the
+    // retranmissions are identical to the original. Also, verify the PTO durations.
+    let mut server = default_server();
+    let mut total_ptos: Duration = Duration::from_secs(0);
+    for i in 1..=3 {
+        let si = server.process(ci.as_ref(), now).dgram().unwrap();
+        assert_eq!(si.len(), server.plpmtu());
+        assert_eq!(
+            server.stats().frame_tx,
+            FrameStats {
+                crypto: i * 2,
+                ack: i,
+                all: i * 3,
+                ..Default::default()
+            }
+        );
+        ci = None;
```

Comment:
```suggestion
        let si = server.process(&ci.take(), now).dgram().unwrap();
        assert_eq!(si.len(), server.plpmtu());
        assert_eq!(
            server.stats().frame_tx,
            FrameStats {
                crypto: i * 2,
                ack: i,
                all: i * 3,
                ..Default::default()
            }
        );
```

Using `take()` is a bit simpler and concise. Haven't tried to compile it. Feel free to ignore.